src/server/middleware/StaticAssetHandler.ts

```diff
@@ -0,0 +1,57 @@
+import { createReadStream } from 'fs';
+import * as mime from 'mime-types';
+import { getLoggerFor } from '../../logging/LogUtil';
+import { APPLICATION_OCTET_STREAM } from '../../util/ContentTypes';
+import { pipeSafely } from '../../util/StreamUtil';
+import type { HttpHandlerInput } from '../HttpHandler';
+import { HttpHandler } from '../HttpHandler';
+import type { HttpRequest } from '../HttpRequest';
+
+/**
+ * Handler that serves static resources on specific paths.
+ */
+export class StaticAssetHandler extends HttpHandler {
+  private readonly assets: Record<string, string>;
+  private readonly logger = getLoggerFor(this);
+
+  /**
+   * Creates a handler for the provided static resources.
+   * @param assets - A mapping from URL paths to file paths.
+   */
+  public constructor(assets: Record<string, string>) {
+    super();
+    this.assets = { ...assets };
+  }
+
+  public async canHandle({ request }: HttpHandlerInput): Promise<void> {
+    if (request.method !== 'GET' && request.method !== 'HEAD') {
+      throw new Error('Only GET and HEAD requests are supported');
+    }
+    if (!(this.getAssetUrl(request) in this.assets)) {
+      throw new Error(`No static resource at ${request.url}`);
+    }
+  }
+
+  public async handle({ request, response }: HttpHandlerInput): Promise<void> {
+    // Determine the asset to serve
+    const filePath = this.assets[this.getAssetUrl(request)];
+    this.logger.debug(`Serving ${request.url} via static asset ${filePath}`);
+
+    // Send the response headers
+    const contentType = mime.lookup(filePath) || APPLICATION_OCTET_STREAM;
+    response.writeHead(200, { 'content-type': contentType });
+
+    // For HEAD, send an empty body
+    if (request.method === 'HEAD') {
+      response.end();
+    // For GET, stream the asset
+    } else {
+      const asset = createReadStream(filePath, 'utf8');
+      pipeSafely(asset, response);
+    }
```

Comment:
Similarly, if you use a (Basic)ResponseWriter here there is an extra failsafe in case there is a problem piping the stream for example.

Reply:
We are testing for error conditions; don't think we could do anything more?
How specific is ResponseWriter to LDP space?

Reply:
The BasicResponseWriter attaches an error listener to the pipe: https://github.com/solid/community-server/blob/693d48b9eb965c4a479e137eea157eb1943b40a9/src/ldp/http/BasicResponseWriter.ts#L36-L39

Regarding the LDP: TargetExtractor takes as input an HttpRequest and returns a ResourceIdentifier, FileIdentifierMapper takes a ResourceIdentifier and returns a file path and content type, and the ResponseWriter takes a statuscode, RepresentationMetadata and Readable. So the ResourceIdentifier and RepresentationMetadata can be seen as links with LDP part, which is why I also didn't have a strong request for this. It just felt like we're doing things in this class that we already solved in other classes.

Reply:
> The BasicResponseWriter attaches an error listener to the pipe:

Thanks; looking at this now, I currently don't see the advantages of adding these components. An error condition (which typically is file not found) will currently also be logged (I think via the Express server logger), so we are covered.